### PR TITLE
Fixing wrong and missing SQL Server type mappings

### PR DESCRIPTION
**Pull request by lajones**
Fix for #2740 - the SQL Server type float was incorrectly mapped to C# float (should be double) and SQL Server type real was incorrectly mapped to C# double (should be float).

Also we were missing a rowversion mapping - we only had its synonym, timestamp.

### CODE REVIEW

File: src/EntityFramework.SqlServer.Design/ReverseEngineering/SqlServerMetadataModelProvider.cs [hunk 1]
```diff
@@ -613,9 +613,10 @@ public static string ConstructIdForCombinationOfColumns([NotNull] IEnumerable<st
             }
 
             if (tableColumn.IsStoreGenerated
+                || tableColumn.DataType == "rowversion"
```
Comment (ErikEJ) [inline src/EntityFramework.SqlServer.Design/ReverseEngineering/SqlServerMetadataModelProvider.cs]: The datatype name returned fro the schema query in all current versions of SQL Server is always "timestamp" and never "rowversion"

File: src/EntityFramework.SqlServer.Design/Utilities/SqlServerTypeMapping.cs [hunk 2]
```diff
@@ -54,6 +54,7 @@ public static class SqlServerTypeMapping
                 //{ "hierarchyid", typeof(TODO) },
                 //{ "sql_variant", typeof(TODO) },
                 //{ "table", typeof(TODO) },
+                { "rowversion", typeof(byte[]) },
```
Comment (ErikEJ) [inline src/EntityFramework.SqlServer.Design/Utilities/SqlServerTypeMapping.cs]: Same comment as above

File: src/EntityFramework.SqlServer.Design/ReverseEngineering/SqlServerMetadataModelProvider.cs (hunk 1 above)
Comment (divega) [inline src/EntityFramework.SqlServer.Design/ReverseEngineering/SqlServerMetadataModelProvider.cs]: And, just to be sure, by "current" in this case you mean "existing"?

Comment (ErikEJ) [inline src/EntityFramework.SqlServer.Design/ReverseEngineering/SqlServerMetadataModelProvider.cs]: Yes, all existing

Comment (lajones) [inline src/EntityFramework.SqlServer.Design/ReverseEngineering/SqlServerMetadataModelProvider.cs]: My mistake. I was making the code consistent with what EF6 did, but it looks like (as usual) Erik is right. I'll update.

**Comment by lajones** [comment]
@ErikEJ @divega - see new commit - updated as requested.

**Comment by divega** [comment]
:shipit:

Is it expected that the E2E.sql is treated by GitHub (or maybe by git?) as binary?

**Comment by ErikEJ** [comment]
Not if the file is saved as ansi cp 1252, but if it is saved as unicode, then yes.

**Comment by lajones** [comment]
Checked in with commit fd723d3

**Pull request closed without merging**